### DISCUSSION TITLE
docs/web/senior-static-server

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#49] [General]
# docs/web/senior-static-server

了解如何在GoFrame框架中配置和使用静态文件服务，包括静态文件目录设置、静态文件服务开启条件、URI与静态目录映射、自定义URI重写规则及跨域配置示例，帮助开发者有效管理和优化项目中的静态资源访问。

https://goframe.org/docs/web/senior-static-server

**Comment by konster** [comment]
如何控制只有登录的用户才可以访问静态文件服务？

**Reply by gqcn** [reply]
可以考虑使用`basic auth`，或者自定义`hook`来控制。

**Comment by konster** [comment]
要是能给个demo就好了



---Original---
From: "John ***@***.***&gt;
Date: Tue, Nov 12, 2024 20:19 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gogf/gf-site] docs/web/senior-static-server (Discussion #49)




 
可以考虑使用basic auth，或者自定义hook来控制。

**Comment by ninjashixuan** [comment]
希望后面能兼容支持 fs.FS 就好了，现在挺常用 embed.FS 的，一些项目我还挺喜欢用 embed.FS 打包进去 vue build 好的文件，一个二进制直接部署。

**Reply by ninjashixuan** [reply]
发现自己用标准库 http.FileServerFS(fs) Wrap 一下也用不了几行代码，好像没太大的必要了。

**Comment by shitiandmw** [comment]
静态文件能支持gzip么

**Reply by gqcn** [reply]
默认不支持，建议前置`Nginx`或者加个`Hook`来实现压缩。
